### ISSUE TITLE
Digital Ocean acme-challenge breaks when over 20 domains 

### ISSUE THREAD

**Issue by Xpyder** [webprofusion/certify#572]
I’m not sure where to submit a bug report for a module (specifically the digital ocean dns auto-validate powershell script: DOcean.ps1)

I dug into the code and the error seems to be that ‘Find-DOZone’ Doesn’t properly support pagination
This breaks the ability to use the script when you have more than 20 domains across all of your certificates

See:
https://docs.digitalocean.com/reference/api/api-reference/#section/Introduction/Links-and-Pagination

For ease of debugging here’s the code to replicate the “check domains” part of Find-DOZone in DOcean.ps1

```
$ApiRoot = 'https://api.digitalocean.com/v2/domains'
$RestParams = @{
    Headers =  @{Authorization='Bearer {your api token}'}
    ContentType = 'application/json'}
$script:UseBasic = @{UseBasicParsing=$True}
(Invoke-RestMethod "$ApiRoot" @RestParams @script:UseBasic).domains|tee -va domains
$domains.count
```

Related post on the forums: https://community.certifytheweb.com/t/digital-ocean-acme-challenge-breaks-when-over-20-domains/1505/8

**Comment by Xpyder** [comment]
Here’s equivalent code to the above that does support the pagination and gathers all the domains into $zones (as used in the script)

```
$ApiRoot = 'https://api.digitalocean.com/v2/domains'
$RestParams = @{
    Headers =  @{Authorization='Bearer {Your API Key}'}
    ContentType = 'application/json'}
$script:UseBasic = @{UseBasicParsing=$True}

		$zones = @()
		$responses = @()
		$request = $ApiRoot
		do {
			try {
				$responses += Invoke-RestMethod "$request" @RestParams @script:UseBasic
			} catch {
				throw
			}
			$zones += ($responses[-1]).domains
			$request = $responses[-1].links.pages.next
		} until (
			($null -eq $request)
		)
```

**Comment by Xpyder** [comment]
I was going to submit a PR here but I'm having trouble finding the scripts\DNS\PoshACME\Plugins folder in this repository

**Comment by webprofusion-chrisc** [comment]
Thanks, we git pull a specific point in time from the Posh-ACME repo when we build our installer, so we don't keep that in our repo. The problem we do have with staying in sync with Posh-ACME is that to integrate the latest changes we need to re-test our general integration with Posh-ACME (which is a workaround as the Posh-ACME providers don't really expect to be used outside of Posh-ACME), and some of the parameters etc may have changed significantly etc.

**Comment by Xpyder** [comment]
ty, I submitted the issue there and they acknowledged it and put it in their queue to be fixed

So I guess the status of this issue for now is waiting on:
 - Posh-ACME to fix it
 - Then Testing review for updated Posh-ACME integration

**Comment by webprofusion-chrisc** [comment]
Yes, Ryan of posh-acme has released the fix (v4.9.0), so it's testing with the latest version of posh acme scripts that's the real holdup. Could be started in about a week depending on other priorities.

**Comment by webprofusion-chrisc** [comment]
Closing. The latest release of Certify The Web (5.5.7) now includes the latest version of Posh-ACME.
